lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -594,7 +618,14 @@ function setupEventListeners(transport) {
     }
   });
 
-  createOrResetTwilioConnection();
+  const { _options, _iceServersStatus } = transport;
+  const { iceServers, onIced } = _options;
+
+  if (_iceServersStatus === 'overrode') {
```

Comment:
this should not part be part of `setupEventListeners` - I suggest we move this call to constructor and also make `createOrResetTwilioConnection` class member. That way, you can clearly make onIced processing just for 1st connection.

Reply:
@makarandp0 `setupEventListeners` is called only once anyway. So, I think this is fine where it is.

Reply:
yes, you are right - its called just once, and also as you mentioned "connecting" state happens just once, so we are good. But I would still move it to constructor - since it feels outof place in `setupEventListeners`

Reply:
@makarandp0 What you're suggesting will require quite a bit of refactoring, so what I'll do is rename the function to `setupTransport()`, so that this function is not just about event listeners.